### DISCUSSION TITLE
Searching through text hyperlinks inside notes

### DISCUSSION THREAD

**Discussion by valdi-1** [freeplane/freeplane#376] [Help]
Hello,

I'm new to the forum...not sure if I'm asking my question in the right place

I'm wondering if there's a way to search NOTES hyperlink values (that is text hyperlinks inside notes)?

Hyperlinks or notes are searchable categories inside "Search box". However, hyperlinks are searched 
only at node level. And notes appear to be searched only at text level - not "looking" into embedded
hyperlink values.

Thanx,

**Comment by EdoFro** [comment]
Hi,
you can use this code in "script filter":

```groovy
def myInput = node.map.root['searchNoteHtml'] ?: (node.map.root['searchNoteHtml']='href="http')
return node.note && node.note.html.toLowerCase().contains(myInput)
```

it will also create an attribute in the root node that you can edit to define what to search.

hope it helps,

edo

**Reply by EdoFro** [reply]
"script filter" is a kind of filter you can choose in the filter toolbar. here is an image:

![image](https://user-images.githubusercontent.com/35700575/169049791-adf29044-18f8-4118-8a78-fabcfcb46e56.png)

**Reply by valdi-1** [reply]
Edo, thank you for your suggestion...

It works... but has two drawbacks: 
   - it modifies the map (by creating an attribute)
   - attribute value has to be manually input each time a search is to be performed

Here's a scenario I have in mind:
   1. Node "A" has a note with embedded text hyperlink to node "B" (using alias which I set for node "B")
   2. Node "B" alias is set to "alias_123"
   3. Node "A's" note hyperlink to node "B" then looks like this: "#at(:~alias_123)"

Here's what I'm looking to accomplish:
   1. Select any node with an alias (in this example it would be node "B")
   2. Invoke search filter to read the value of node "B's" alias (i.e.  "alias_123")
   3. Search mindmap's notes for a full hyperlink value;  "#at(:~alias_123)"
       (programmatically this would mean combining three substrings: "#at(:~" + selected node's alias value + ")"
   4. Highlight (or select) nodes with notes - if found - containing hyperlinks matching the above search criteria

This way I wouldn't have to deal with a modified map nor manual input each time I'd need to perform a search.

Thanks for your time in advance...


p.s. Perhaps such solution could be added to search tool's  enhancement "wish" list?

**Reply by valdi-1** [reply]
Just noticed the editor placed "strike through" part of my submitted reply - as if it were to be ignored...

Not sure why it did it, but it was NOT meant to be ignored
